Sources/ApodiniDeploy/ApodiniDeployExporterConfiguration.swift

```diff
@@ -11,29 +11,28 @@ import Apodini
 import ApodiniDeployBuildSupport
 import ApodiniDeployRuntimeSupport
 
-
-public struct ApodiniDeployConfiguration: Apodini.Configuration {
-    struct StorageKey: Apodini.StorageKey {
-        typealias Value = ApodiniDeployConfiguration
-    }
-    
+struct ApodiniDeployExporterConfiguration {
```

Comment:
I think this is a breaking change for the deployment provider logic and probably results in failed test cases if we would enable the deployment tests (CCL @lukaskollmer)

I think it might be reasonable to add the functionality of subcommands of the Swift Argument parser to Apodini: https://github.com/apple/swift-argument-parser/blob/main/Documentation/03%20Commands%20and%20Subcommands.md

I would suggest to extend `Apodini.Configuration` with an additional property:
```swift
import ArgumentParser


/// `Configuration`s are used to register services to Apodini.
/// Each `Configuration` handles different kinds of services.
public protocol Configuration {
    var subcommands: [ParsableCommand.Type] { get }
    
    /// A method that handles the configuration of a service which is called by the `main` function.
    ///
    /// - Parameter
    ///    - app: The `Vapor.Application` which is used to register the configuration in Apodini
    func configure(_ app: Application)
}

extension Configuration {
    public var subcommands: [ParsableCommand.Type] {
        []
    }
}

// ...

extension Array: Configuration where Element == Configuration {
    public var subcommands: [ParsableCommand.Type] {
        var subcommands: [ParsableCommand.Type] = []
        forEach {
            subcommands.append(contentsOf: $0.subcommands)
        }
        return subcommands
    }
    
    public func configure(_ app: Application) {
        forEach {
            $0.configure(app)
        }
    }
}
```

The logic currently found in the WebService could move to a `Start` `ParsableCommand` that is the default parsable command in the WebService:
```swift

public extension WebService {
    var configuration: CommandConfiguration {
        CommandConfiguration(
            abstract: "A utility for performing maths.",
            subcommands: [Start.self] + self.configuration.subcommands,
            defaultSubcommand: Start.self
        )
    }
}

struct Start: ParsableCommand {
    // ...
}
```


This might be relevant for 
- `Apodini Deploy`: @hendesi & @lukaskollmer The deployment provider would add a deploy subcommand and then e.g. an export sub subcommand that triggers the export functionality so you can call it like this: `MyWebService deploy export --mode ... --file ... --node` and so on. Adding this to the main web service using Arguments doesn't really make sense but enabling configurations to add this as subcommands is definitely a welcome thing. Please get in contact with @lukaskollmer as this current change definitely breaks something in the deployment providers and we need to make sure they still work after the PR. @hendesi will continue this work with his deployment providers and I can imagine that these subcommand structure might also become helpful for him as we have discussed incorporating deployment provider right in the main web service binary. You could then deploy your web service to AWS Lambda e.g. executing the web service with `MyWebService deploy awslambda --username ...`
- `Apodini Database`: Relevant for @hendesi : It would be good to have a command like `MyWebService database migrate` or `MyWebService database revert` to migrate and revert database schemas without starting the actual web service. The command could then exit after the migration/revert with a success exit code and then migrating database changes is easier then starting up the web service to run the migrations and currently reverting a migration is not even possible.

Reply:
First of all, I really like the idea to have different subcommands that trigger various functionality in Apodini (eg. migrating DB without startup of whole Apodini WebService)!

However, there are a few problems I would like to mention:

- You are declaring the `var configuration: CommandConfiguration` in the `WebService` as non-static and as a computed property. However, a look into the documentation shows that the configuration first needs to be static (the easy thing to solve) and second it MUST be a stored property (i tried it out myself, a computed property apparently isn't recognized by the ArgumentParser and therefore the `run` method in `Start` is never called, only the original `run` in the `WebService`). A stored property in a protocol extension isn't possible, therefore I guess we have a problem here.
![image](https://user-images.githubusercontent.com/25406915/121821616-59e2d580-cc9a-11eb-88ab-95514febe1b7.png)
I guess the developers have to specify a `CommandConfiguration` themselves in the `WebService` struct, a very unpleasant solution. 

- Name conflict between the new `configuration: CommandConfiguration` and the old `configuration: Configuration`, we would have to rename the `configuration: Configuration` to something like `ExporterConfiguration`.
- What logic from the `WebService` would move to the `Start` exactly? I would argue for as little as possible, so just the `run` method required by the `ArgumentParser` as well as the `start` functions from Apodini. So, ground rules established. Now, to make the CLI-Arguments in the Start CMD configurable by the user (clarification: if we execute `webserice start --test=abc`, the parameter is only injected into the `Start` struct), the `Start` struct must actually be a protocol. I see two problems here: First, if the `configuration: Configuration` stays in the `WebService`, how would the injected arguments in `Start` be used in the `configuration: Configuration` in the `WebService`? Second, if `Start` is a protocol, the actual struct instantiation can be named arbitrarily. How do we reference this struct then in the `CommandConfiguration`? (If we require the developer to implement a `CommandConfiguration` manually in the WebService as well as a struct `StartCMD` that conforms to `Start` which conforms to `ParsableCommand` and provides a default implementation for the `run` function, then it would be possible)
- Another option would be to move the `configuration: Configuration` to `Start` (so conformance to the `ConfigurationCollection)`. In my opinion a very unpleasant solution. Now it would be possible to use the passed CLI arguments in the configuration etc. But again, how do we specify the actual `Start` struct in the `CommandConfiguration` in the `WebService` then? And, I think it breaks LOTS of things if we don't have a configuration in the `WebService`. I also think that this would be the completely wrong direction, it really doesn't make sense to specify the configuration in the `Start` struct and not in the `WebService`
- Probably a lot more problems I'm not seeing right away

It's very well possible that there is a very easy solution here that I'm not seeing. If that's the case, please let me know ASAP :)

Reply:
I will implement this feature in a separate PR once this one is merged